evalml/automl/engine/engine_base.py

```diff
@@ -52,6 +55,65 @@ def evaluate_batch(self, pipelines):
             list (int): A list of the new pipeline IDs which were created by the AutoML search.
         """
 
+    @abstractmethod
+    def train_batch(self, pipelines):
+        """Train a batch of pipelines using the current dataset.
+
+        Arguments:
+            pipelines (list(PipelineBase)): A batch of pipelines to fit.
+        Returns:
+            list(PipelineBase): List of fitted pipelines
+        """
+        if self.X_train is None or self.y_train is None:
```

Comment:
Great question. I prefer if the engines don't hold on to the data but my understanding is that we originally designed it this way so that the `ParallelEngine` would be able to send the data to the dask workers only once. If you think we don't need that for the `ParallelEngine` then I'm all for it but it probably belongs in #1914 ?

Reply:
Yea, no need to address it right now.  This can wait until the redesign